pkg/hubble/parser/threefour/parser.go

```diff
@@ -534,6 +534,22 @@ func decodeTrafficDirection(srcEP uint32, dn *monitor.DropNotify, tn *monitor.Tr
 			}
 			return pb.TrafficDirection_INGRESS
 		}
+
+		// Try inferring direction from observation point given the UNKNOWN reason/endpoint.
+		// Ignore Trace{From,To}Stack and Trace{From,To}Proxy since they can be used for
+		// both ingress and egress traffic.
+		switch tn.ObsPoint {
+		case monitorAPI.TraceToLxc,
+			monitorAPI.TraceToOverlay,
+			monitorAPI.TraceToCrypto,
+			monitorAPI.TraceToNetwork:
+			return pb.TrafficDirection_EGRESS
+		case monitorAPI.TraceFromLxc,
+			monitorAPI.TraceFromOverlay,
+			monitorAPI.TraceFromCrypto,
+			monitorAPI.TraceFromNetwork:
+			return pb.TrafficDirection_INGRESS
+		}
```

Comment:
guessing ingress/egress based on the observation point make sense to me intuitively, I wonder why we didn't do it sooner cc @gandro who has more context around this codepath than I do.

Also question: is `TRACE_TO_LXC → EGRESS` and `TRACE_TO_LXC → INGRESS` correct here? Isn't it the other way around?

Reply:
> guessing ingress/egress based on the observation point make sense to me intuitively, I wonder why we didn't do it sooner cc @gandro who has more context around this codepath than I do.

We did consider this, but we ended up not doing it because I'm afraid it's not correct. The traffic direction field is supposed to show the direction of the **connection** the packet belongs to, not the direction of the packet itself. 

So a reply packet (egressing the node) for an ingress connection should still be classified as ingress. I'd rather leave the traffic direction field empty here.

Reply:
> The traffic direction field is supposed to show the direction of the connection the packet belongs to, not the direction of the packet itself.

That explains a lot, thanks for clarifying it :pray: 
Then, also for  `TRACE_{FROM,TO}_CRYPTO` events, I'd let the parser try inferring it from the trace reason and/or the other info known, removing this portion of code (and commits) from the PR.